confgenerator/otel/modular.go

```diff
@@ -143,13 +146,20 @@ func (c ModularConfig) Generate() (string, error) {
 		"service":    service,
 	}
 
-	resourceDetectionProcessors := map[ResourceDetectionMode]Component{
+	resourceProcessors := map[ResourceDetectionMode]Component{
```

Comment:
I would prefer keeping this named `resourceDetectionProcessors` (especially since you're going to switch back to a resource detection processor when it's supported upstream, right?)